### PR TITLE
create lookup selector widget

### PR DESCRIPTION
**Pull request by ChristosAsvestopoulos**
**working POC for lookup selection:**

From layout settings we have something like this:
```
	"lookup-name": "productCode",
	"lookup-filter-terms": [
		{
			"key": "caseType",
			"value": "Certification.DepositInfo"
		}
	]
```
lookup-name is used by the factory to create the appropriate instance type of ILookupService
lookup-filter-terms are used to pass extra parameters to implementations of ILookupService.

A client can easily register an implementation of ILookupService, using:
`services.AddLookupService<MockOccupationCodesLookupService>(LookupName.OccupationCode);`

We can also have:
```
	"lookup-name": "productCode",
	"lookup-filter-terms": [
		{
			"key": "caseType",
			"value": "Certification.DepositInterest"
		}
	],
	"lookup-filter-fields": [
		"date"
	]
```
where lookup-filter-fields contains the form field(s) that the current lookup is dependent on.

---
The widget always sends a customerId but that can simply be ignored from client's implementations of ILookupService.

### CODE REVIEW

File: src/Indice.Features.Cases.AspNetCore/Interfaces/ILookupService.cs [hunk 1]
```diff
@@ -14,6 +14,6 @@ public interface ILookupService
         /// <param name="lookupName">The lookup name.</param>
         /// <param name="searchValues">Any search values to filter the results.</param>
         /// <returns></returns>
-        Task<ResultSet<LookupItem>> Get(string lookupName, string searchValues = null);
+        Task<ResultSet<LookupItem>> Get(string lookupName, SearchValues searchValues = null);
```
Comment (ChristosAsvestopoulos) [inline src/Indice.Features.Cases.AspNetCore/Interfaces/ILookupService.cs:17]: the type of searchValues parameter should be "client agnostic".

**Comment by ChristosAsvestopoulos** [comment]
should also add a changelog for Cases UI

File: src/Indice.Features.Cases.AspNetCore/Controllers/LookupController.cs [hunk 2]
```diff
@@ -38,9 +38,9 @@ public class LookupController : ControllerBase
         [Produces(MediaTypeNames.Application.Json)]
         [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(ResultSet<LookupItem>))]
         [ProducesResponseType(StatusCodes.Status404NotFound, Type = typeof(ProblemDetails))]
-        public async Task<IActionResult> GetLookup([FromRoute] string lookupName, [FromQuery] string searchValues = null) {
-            var occupations = await _lookupService.Get(lookupName, searchValues);
-            return Ok(occupations);
+        public async Task<IActionResult> GetLookup([FromRoute] string lookupName, [FromQuery] SearchValues searchValues = null) {
+            var lookupItems = await _lookupService.Get(lookupName, searchValues);
```
Comment (ChristosAsvestopoulos) [inline src/Indice.Features.Cases.AspNetCore/Controllers/LookupController.cs:42]: use factory pattern here

File: src/Indice.Features.Cases.App/src/app/shared/ajsf/json-schema-frameworks/tailwind-framework/product-selector-widget/product-selector-widget.component.ts [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+  ) { }
+
+  ngOnInit(): void {
+    this.options = this.layoutNode.options || {};
+    this.dependentField = this.options['dependentField'];
+    if (this.dependentField) {
+      this.dependentFieldCurrentValue = this.jsf.data[this.dependentField];
+    }
+    let lookupName = this.options['lookup-name'] ?? this.controlName;
+    let productFamily = this.options['productFamily'] ?? 'Account';
```
Comment (ChristosAsvestopoulos) [inline src/Indice.Features.Cases.App/src/app/shared/ajsf/json-schema-frameworks/tailwind-framework/product-selector-widget/product-selector-widget.component.ts:44]: make it general

File: src/Indice.Features.Cases.AspNetCore/Interfaces/ILookupService.cs (hunk 1 above)
Comment (ChristosAsvestopoulos) [reply src/Indice.Features.Cases.AspNetCore/Interfaces/ILookupService.cs]: Firstly, we should use ListOptions and, secondly, we should use something like `{'name':'...', 'value':'...'}`.

File: src/Indice.Features.Cases.AspNetCore/Services/NoOpServices/NoOpLookupService.cs [hunk 4]
```diff
@@ -1,12 +1,15 @@
 ﻿using Indice.Features.Cases.Interfaces;
+using Indice.Features.Cases.Models;
 using Indice.Features.Cases.Models.Responses;
 using Indice.Types;
 
 namespace Indice.Features.Cases.Services.NoOpServices
 {
     internal class NoOpLookupService : ILookupService
```
Comment (ChristosAsvestopoulos) [inline src/Indice.Features.Cases.AspNetCore/Services/NoOpServices/NoOpLookupService.cs:8]: remove it?

File: src/Indice.Features.Cases.App/src/app/shared/ajsf/json-schema-frameworks/tailwind-framework/lookup-selector-widget/lookup-selector-widget.component.ts [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+  templateUrl: './lookup-selector-widget.component.html'
+})
+export class LookupSelectorWidgetComponent implements OnInit {
+  formControl: any;
+  controlName: string | undefined;
+  controlValue: string | undefined;
+  /**
+  * Holds the name(s) of the field(s) that the widget is dependent on.
+  */
+  lookupFilterFields: string[] | undefined;
```
Comment (ChristosAsvestopoulos) [inline src/Indice.Features.Cases.App/src/app/shared/ajsf/json-schema-frameworks/tailwind-framework/lookup-selector-widget/lookup-selector-widget.component.ts:20]: private

File: src/Indice.Features.Cases.App/src/app/shared/ajsf/json-schema-frameworks/tailwind-framework/lookup-selector-widget/lookup-selector-widget.component.ts [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+import { Subject } from "rxjs";
+import { take, takeUntil } from "rxjs/operators";
+import { CaseDetailsService } from "src/app/core/services/case-details.service";
+import { CaseDetails, CasesApiService, FilterTerm, LookupItemResultSet } from "src/app/core/services/cases-api.service";
+
+@Component({
+  selector: 'app-lookup-selector-widget',
+  templateUrl: './lookup-selector-widget.component.html'
+})
+export class LookupSelectorWidgetComponent implements OnInit {
```
Comment (ChristosAsvestopoulos) [inline src/Indice.Features.Cases.App/src/app/shared/ajsf/json-schema-frameworks/tailwind-framework/lookup-selector-widget/lookup-selector-widget.component.ts:13]: use methods to reduce duplicated code

**Review by dkarkanas — APPROVED** [review]

File: src/Indice.Features.Cases.AspNetCore/Services/NoOpServices/NoOpLookupService.cs (hunk 4 above)
Comment (ChristosAsvestopoulos) [reply src/Indice.Features.Cases.AspNetCore/Services/NoOpServices/NoOpLookupService.cs:8]: no, it is needed for instantiating the factory

**Pull request merged**